relay-server/src/actors/outcome.rs

```diff
@@ -304,6 +306,9 @@ pub struct TrackRawOutcome {
     /// The source of the outcome (which Relay sent it)
     #[serde(default, skip_serializing_if = "Option::is_none")]
     source: Option<String>,
+    /// The event's data category.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub category: Option<DataCategory>,
```

Comment:
We're good with `TrackOutcome` having a required category, but does *this* one (on `TrackRawOutcome`) need to be optional? It seems that `TrackRawOutcome` is being used, in `relay-server/src/endpoints/outcomes.rs`, to deserialize outcomes from upstream Relays. If those Relays are self-hosted ones that aren't running the latest version, they might be sending JSON that lacks the `category` attribute, and we want to tolerate that, right? (Credit to @JoshFerge for the catch.)

I checked in d117009 on the hypothesis that this is correct. I'll revert it otherwise.